_posts/2024-04-04-cheriot-supply-chain.markdown

```diff
@@ -0,0 +1,189 @@
+These are like global variables but where the owning compartment does not have access to the object, only to a sealed capability (opaque pointer) to them.
+These are used to bootstrap various things.
+For example, if you want to allocate memory then you must present an allocator capability to the allocator compartment.
+This contains the quota that determines the maximum amount that you can allocate.
+Similarly, to create a network connection you must present the network API compartment with a sealed capability authorising you to connect to a particular host.
+
+The core of the RTOS uses sealing to remove the scheduler from the trusted computing base (TCB) for confidentiality and integrity.
+On context switch, the switcher spills registers and then invokes the scheduler with a sealed capability to the register-save area.
+The scheduler then hands back a sealed capability to a register-save area.
+The scheduler cannot see the state for the interrupted thread and the switcher can detect whether the capability that it's given for a new save area is a valid one.
+
+The cross-compartment call mechanism in CHERIoT is also built on top of sealing.
+Sealing is one of the most important parts of a CHERI system because it's the thing that enables all of these compartmentalisation-related abstractions.
+
+## Compartment boundaries become auditing boundaries
+
+Once you have defined security boundaries around third-party code, you can use them for auditing.
+The CHERIoT ABI and compartment model were designed with this in mind.
+A trivial compartment has just two capabilities, one to its code and one to its data.
+Such a compartment is not useful: it cannot ever run.
+A useful compartment needs to export at least one entry point and typically import entry points from other compartments.
+
+To be able to invoke any entry point in another compartment, the loader must give the compartment a sealed capability for that entry point.
+For library calls, this is also true, though with a special kind of sealed capability that the hardware unsealed on a jump (a sealed entry, or 'sentry' capability).
+
+The loader will provision capabilities to compartments based on metadata filled in by the linker.
+This means that the linker knows (among other things), for every compartment:
+
+ - What entry points it exposes.
+ - Which of those entry points run with interrupts enabled or disabled.
+ - Which entry points in other compartments (or libraries) it calls.
+ - What static sealed objects does it have (including the contents of those objects and which compartment owns the capability that authorises unsealing them).
+ - What the exact contents of the code is for each compartment.
+ - Which memory-mapped I/O (MMIO) regions it can access.
+
+If the linker doesn't know one of these things, it can't tell the loader to provide the capabilities that authorise it.
+CHERI systems fail closed: if you don't have a capability that authorises a thing, you cannot do the thing.
```

Comment:
This sentence seems like a bit of a non-sequitur. I think it does make sense but it took me a second to figure out the context.